ndcube/ndcube.py

```diff
@@ -792,3 +801,90 @@ def plot(self, *args, **kwargs):
                 "no default plotting functionality is available.")
 
         return self.plotter.plot(*args, **kwargs)
+
+    def _new_instance_from_op(self, new_data, new_unit):
+        # This implicitly assumes that the arithmetic operation does not alter
+        # the WCS, mask, or uncertainty
+        new_cube = type(self)(new_data,
+                              unit=new_unit,
+                              wcs=self.wcs,
+                              mask=self.mask,
+                              meta=self.meta,
+                              uncertainty=self.uncertainty)
+        if self.extra_coords is not None:
+            new_cube._extra_coords = deepcopy(self.extra_coords)
+        return new_cube
+
+    def __neg__(self):
+        return self._new_instance_from_op(-self.data, self.unit)
+
+    def __pow__(self, value):
```

Comment:
I think this needs to propagate uncertainties.  See #450 for my current attempt on how to do this.  If this is too much to agree on by Friday, I suggest not implementing `__pow__` until we've agreed an implementation for uncertainty propagation.

Reply:
Oh yes, good point! I think at this stage, we should just explicitly check that `self.uncertainties is None` and if not, throw a `NotImplemented`.

Reply:
I actually disagree (in a friendly tone :) ).  I think if we give people something that returns `NotImplementedError` in some cases it may be confusing and frustrating.  Then once we do implement it fully, it may take some time and effort to change the pwerception that power doesn't work properly.  If we don't release it, `NDCube` hasn't gotten any worse and when we do release `pow` it should be a much more positive users experience.  I'm not aware that we have any users asking explictly for `pow` at the moment...

Reply:
I don't know if @wtbarnes intended to write `NotImplemented`, which is different from `NotImplementedError`.  I think situationally returning `NotImplemented` depending on the particulars of an instance rather than just the types is a bad idea.

Reply:
I specifically wrote `NotImplemented` and not `NotImplementedError`. My only reasoning for this is this is what we return in the arithmetic operators on Map when operations are not supported. I recall that we had a discussion about which to return when the Map arithmetic operators were implemented and the consensus was `NotImplemented`. However, I've lost the thread on what exactly the reasoning was. I have no opinion here but we should probably be consistent across Map and NDCube.

Reply:
The problem with returning `NotImplemented` in only a subset of situations of using a particular operator (i.e., only when `self.certainties` is `False`) is that there is no way to provide a message to the user explaining why it doesn't work for that particular instance.  It'll simply report that there's no supported operator for those two types, which would actually be a lie.  `NotImplemented` is really intended for an operator that will *never* work given the types.

Reply:
Ah ok. Well that's certainly not the case here then (and also not the case in Map either). I would propose then that we actually return `NotImplementedError` and provide the reasoning in the error message.

Reply:
I think if we are going to support uncertainties, we should be making use of the error propagation provided by astropy; https://docs.astropy.org/en/stable/api/astropy.nddata.NDUncertainty.html#astropy.nddata.NDUncertainty.propagate

Reply:
I'm struggling with how exactly to deal with this, given that `uncertainty` is not even required to adhere to a particular type. If we do not want to support `__pow__` for only cubes *without* uncertainties, then I think it makes sense to just drop it for now.

Reply:
If you set uncertainty to an array it gets converted to `UnknownUncertainty` by `NDCube`,s parent class instantiator.  And `UnknownUncertainty` prevents error propagation.  So you're right in that sense that error propagation is not always possible.  So I think the solution is to check the type.  If `uncertainty` is `UnknownUncertainty` or `None`, then drop uncertainties.  Otherwise `uncertainty` must be a valid uncertainty type that supports propagation.

As I said before, I think it's reasonable to drop `__pow__` from this PR and save it for its own where these issues do not hold up the other arithmetic operations.

Reply:
> Ah ok. Well that's certainly not the case here then (and also not the case in Map either). I would propose then that we actually return `NotImplementedError` and provide the reasoning in the error message.

If we are going to deviate from `Map` here then I would like to change this in `Map` as well, so that we don't end up slit brained (also it might generate more discussion on a sunpy PR).